docs/Architecture.md

```diff
@@ -2,13 +2,37 @@
 
 ## CLI Architecture
 
--   CLI usage starts in `bin/tslint-to-eslint-config`, which immediately calls `src/cli/main.ts`.
--   CLI settings are parsed and read in `src/cli/runCli.ts`.
--   Application logic is run by `src/conversion/convertConfig.ts`.
+1.  CLI usage starts in `bin/tslint-to-eslint-config`, which immediately calls `src/cli/main.ts`.
+2.  CLI settings are parsed and read in `src/cli/runCli.ts`.
+3.  Application logic is run by `src/conversion/convertConfig.ts`.
 
-## Dependencies
+## Configuration Conversion
 
-Methods are created using a variant of [Dependency Injection](http://en.wikipedia.org/wiki/Dependency_Injection).
-Any method with dependencies that should be stubbed out during tests, such as file system bindings, logging, or other methods,
-takes in a first argument of name `dependencies`.
-Its dependencies object is manually created in `src/cli/main.ts` and bound to the method with `bind`.
+Within `src/conversion/convertConfig.ts`, the following steps occur:
+
+1. Existing configurations are read from disk
+2. TSLint rules are converted into their ESLint configurations
+3. ESLint configurations are simplified based on extended ESLint presets
+4. The simplified configuration is written to the output config file
+5. A summary of the results is printed to the user's console
+
+### Rule Converters
+
+Each TSLint rule should output at least one ESLint rule as the equivalent.
+"Converters" for TSLint rules are located in `src/rules/converters/`, and keyed under their names by the map in `src/rules/converters.ts`.
+
+Each converter for a TSLint rule takes an arguments object for the rule, and returns an array of objects containing:
+
+-   `ruleName`: Equivalent ESLint rule name that should be enabled
+-   `ruleArguments`: Any arguments for that ESLint rule
```

Comment:
Converters can also have `notices`, right?

Reply:
Oh yes that's correct! Will add..